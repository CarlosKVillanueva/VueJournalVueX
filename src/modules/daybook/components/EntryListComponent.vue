<template>
	<div class="entry-list-container">
		<div class="px-2 pt-2">
			<input type="text" class="form-control" placeholder="Buscar entradas" v-model="term">
		</div>
	</div>

	<div class="entry-scroll-area">
		<EntryComponent v-for="entry in entriesByTerm" :key="entry.id" :entry="entry"/>
	</div>
</template>

<script>
import { defineAsyncComponent } from "vue"
import { mapGetters } from "vuex";


export default {
	components:{
		EntryComponent: defineAsyncComponent(() => import('./EntryComponent'))
	},
	computed: {
		...mapGetters('journal', ['getEntriesByTerm']),
		entriesByTerm() {
			return this.getEntriesByTerm( this.term )
		},
	},
	data() {
		return {
			term: ''
		}

	},
}
</script>

<style lang="scss" scoped>

input {
	height: 25px;
}
.entry-list-container {
	//border-right: 1px solid #2c3e50;
	height: calc( 100vh - 870px );
}
.entry-scroll-area {
	height: calc(100vh - 110px);
	overflow: scroll;
}
</style>