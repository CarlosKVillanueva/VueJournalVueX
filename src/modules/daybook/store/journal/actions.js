export const loadEntries = async ( /*{ commit }*/ ) => {

}
export const updateEntrry = async ( /*{ commit }*/ ) => {

}

export const createEntrry = async ( /*{ commit }*/ ) => {

}