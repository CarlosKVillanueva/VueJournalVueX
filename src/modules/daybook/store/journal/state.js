export default () => ({
    isLoading: true,
    entries: [
        {
            id: '1',
            date: new Date().toDateString(),
            text: 'All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, ' +
                'making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, ' +
                'combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. ' +
                'The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.A' +
                'll the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, ' +
                'making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, ' +
                'combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. ' +
                'The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.',
            picture: null,
        },
        {
            id: '2',
            date: new Date().toDateString(),
            text: 'Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for \'lorem ipsum\' will uncover many web sites still in their infancy',
            picture: null,
        },
        {
            id: '3',
            date: new Date().toDateString(),
            text: 'Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries',
            picture: null,
        },
    ]
})
