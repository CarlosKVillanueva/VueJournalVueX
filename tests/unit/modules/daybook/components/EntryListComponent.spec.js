import { shallowMount } from "@vue/test-utils"
import EntryListComponent from "@/modules/daybook/components/EntryListComponent"
import { createStore } from "vuex"
import journal from "@/modules/daybook/store/journal"
import { journalState } from "../../../mocks/test-journal-state"


const createVueXStore = ( initialState ) => createStore( {
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
} )
describe( 'EntryList Suite Test', () => {

    const store = createVueXStore( journalState )
    const mockRouter = {
        push: jest.fn()
    }

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()

        wrapper = shallowMount( EntryListComponent, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        } )
    })


    test('Snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()

    });
    
    test('Should call getEntriesByTerm without args and show 2 entries', () => {

        expect( wrapper.findAll( 'entry-stub' ).length ).toBe( 2 )
    });
    
    test('Must call getEntriesByTerm and filter entries', async () => {
        const input = wrapper.find( 'input' )
        await input.setValue( 'deportivo' )

        expect( wrapper.findAll( 'entry-stub' ).length ).toBe( 1 )
    });
    
    test('Button must redirect to /new', () => {
        wrapper.find('button').trigger('click')
        expect( mockRouter.push).toHaveBeenCalledWith({name: 'entry', params: { id:'new' }})
    });
} );