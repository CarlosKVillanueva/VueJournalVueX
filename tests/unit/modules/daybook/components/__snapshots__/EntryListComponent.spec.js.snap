// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList Suite Test Snapshot 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entradas"></div>
</div>
<div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
<div class="entry-scroll-area">
  <entry-component-stub entry="[object Object]"></entry-component-stub>
  <entry-component-stub entry="[object Object]"></entry-component-stub>
</div>
`;
